src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import TabList from './components/TabList';
+import Tab from './components/Tab';
+import TabPanel from './components/TabPanel';
+
+class Tabs extends Component {
+  static propTypes = {
+    initialSelectedIndex: PropTypes.number,
+    items: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        tab: PropTypes.node.isRequired,
+        panel: PropTypes.node.isRequired
+      })
+    ).isRequired
+  };
+
+  state = { selectedIndex: 0 };
+
+  handleChange = selectedIndex => this.setState({ selectedIndex });
+
+  static getDerivedStateFromProps(props, state) {
+    if (
+      props.initialSelectedIndex &&
```

Comment:
Hm, thinking. Is it really initial if you can change it via updating the prop? I know the problem you're solving here though. You don't want to just copy props to state and have them be out of sync.

Reply:
Yea, I had a different functionality when naming this. I am not sure whether to even keep it now, but will rename it at least. Thanks for pointing it out!